concert/cloudformation.choice-aggregator.yml

```diff
@@ -0,0 +1,276 @@
+      ExtendedS3DestinationConfiguration:
+        BucketARN: !GetAtt S3Bucket.Arn
+        BufferingHints:
+          IntervalInSeconds: 60
+          SizeInMBs: 50
+        CloudWatchLoggingOptions:
+          Enabled: true
+          LogGroupName: !Ref KinesisLogGroup
+          LogStreamName: !Ref KinesisFirehoseStreamLogStream
+        CompressionFormat: GZIP
+        EncryptionConfiguration:
+          KMSEncryptionConfig:
+            AWSKMSKeyARN: !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:alias/aws/kinesis
+        Prefix: YYYY/MM/DD/HH
+        RoleARN: !GetAtt KinesisFirehoseStreamRole.Arn
+
+  KinesisAnalyticsApplicationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub ${AWS::Region}-${Environment}-enchanted-brain-kinesis-analytics-v2
+      Path: !Sub /${AWS::Region}/${Environment}/enchanted-brain/
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: kinesisanalytics.amazonaws.com
+            Action: sts:AssumeRole
+      Policies:
+        - PolicyName: !Sub ${AWS::Region}-${Environment}-enchanted-brain-kinesis-analytics
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - firehose:DescribeDeliveryStream
+                  - firehose:Get*
+                Resource: !GetAtt KinesisFirehoseStream.Arn
+              - Effect: Allow
+                Action: lambda:InvokeFunction
+                Resource: !GetAtt AggregateWriterFunction.Arn
+              - Effect: Allow
+                Action: logs:PutLogEvents
+                Resource: !Sub arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/kinesisfirehose/${Environment}-enchanted-brain-kinesis:log-stream:${KinesisAnalyticsLogStream}
+
+  KinesisAnalyticsApplication:
+    Type: AWS::KinesisAnalyticsV2::Application
+    Properties:
+      ApplicationName: !Sub ${Environment}-enchanted-brain-aggregate
+      RuntimeEnvironment: SQL-1_0
+      ServiceExecutionRole: !GetAtt KinesisAnalyticsApplicationRole.Arn
+      ApplicationConfiguration:
+        ApplicationCodeConfiguration:
+          CodeContent:
+            TextContent: !Sub
+              - |
+                CREATE OR REPLACE STREAM ${DestinationStream}
+                (
+                  CHOICE_TIME   TIMESTAMP,
+                  CHOICE_TYPE   VARCHAR(32),
+                  CHOICE_SUM    DOUBLE,
+                  CHOICE_COUNT  int
+                );
+                CREATE OR REPLACE PUMP SUM_PUMP AS
+                  INSERT INTO ${DestinationStream}
+                  SELECT STREAM CHOICE_TIME,
+                    CHOICE_TYPE,
+                    SUM(CAST(CHOICE AS DOUBLE)) AS choiceSum,
+                    COUNT(*) AS choiceCount
+                  FROM ${SourceStreamPrefix}_001 AS SOURCE
+                  WHERE CHOICE_TYPE NOT LIKE 'CHOICE_COLOR'
```

Comment:
If you wanted to not use LIKE, you could do `<>`

Reply:
IMO, `NOT LIKE` is a little bit more legible. If you're okay with it, will just leave it as is,

Reply:
:+1: